tests/core/test_registry.py

```diff
@@ -0,0 +1,64 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pytest
+import torch
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from torch import nn
+
+from flash.core.registry import FlashRegistry
+
+
+def test_registry(tmpdir):
+
+    backbones = FlashRegistry("backbones")
+
+    @backbones.register_function()
```

Comment:
could we directly do like this ?
```python
@flash.backbone
def my_model(nc_input=5, nc_output=6):
  return nn.Linear(nc_input, nc_output), nc_input, nc_output

```

Reply:
I like Edgar's idea 👍

Reply:
@edgarriba I don't think we should do this. This is just for testing and doesn't specify the task the registry belongs to.

Reply:
Why not Justus? For example:

Instead of:
```python
@BACKBONES_REGISTRY.register_function(name="simclr-imagenet", namespace="vision", package="bolts")
```
have:
```python
@flash.backbone(name="simclr-imagenet", namespace="vision", package="bolts")
```

or

```python
@flash.backbone
def simclr_imagenet():
    ...
```

Which could use the function name and use it automatically, useful when there is no metadata

Reply:
Not a huge fan of this. I prefer to stick the current API and modify it later when we have more than backbones.

Reply:
Done !